crates/ruff_python_parser/tests/snapshots/invalid_syntax@expressions__parenthesized__parenthesized.py.snap

```diff
@@ -0,0 +1,74 @@
+---
+source: crates/ruff_python_parser/tests/fixtures.rs
+input_file: crates/ruff_python_parser/resources/invalid/expressions/parenthesized/parenthesized.py
+---
+## AST
+
+```
+Module(
+    ModModule {
+        range: 0..125,
+        body: [
+            Expr(
+                StmtExpr {
+                    range: 66..70,
+                    value: Starred(
+                        ExprStarred {
+                            range: 67..69,
+                            value: Name(
+                                ExprName {
+                                    range: 68..69,
+                                    id: "x",
+                                    ctx: Load,
+                                },
+                            ),
+                            ctx: Load,
+                        },
+                    ),
+                },
+            ),
+            Expr(
+                StmtExpr {
+                    range: 119..120,
+                    value: Name(
+                        ExprName {
+                            range: 119..120,
+                            id: "x",
+                            ctx: Load,
+                        },
+                    ),
+                },
+            ),
+            Expr(
+                StmtExpr {
+                    range: 124..125,
+                    value: NumberLiteral(
+                        ExprNumberLiteral {
+                            range: 124..125,
+                            value: Int(
+                                1,
+                            ),
+                        },
+                    ),
+                },
+            ),
+        ],
+    },
+)
+```
+## Errors
+
+  |
+1 | # Starred expression isn't allowed in a parenthesized expression.
+2 | (*x)
+  |  ^^ Syntax Error: starred expression cannot be used here
+3 | 
+4 | # Unparenthesized named expression is allowed.
+  |
+
+
+  |
+4 | # Unparenthesized named expression is allowed.
+5 | x := 1
+  |   ^^ Syntax Error: Expected a statement
+  |
```

Comment:
You might want to move this into its own file (or add another test in a seperate file to test both the recovery and the unparenthesized named expression case) because it seems that the actual and expected diagnostics don't.

Reply:
Wait, it should say "Unparenthesized named expression is _not_ allowed"

Reply:
Actually, this is correct. It's just the way parser is recovering. It's similar to what we talked about yesterday for `yield` expression. Here, the grammar doesn't allow named expression at the statement level which means the first statement is `x` and the parser moves on to the next statement. It's starting with `:=` which cannot start any statement. 

I'll add this to the error recovery tracking issue. It is easy to solve, just that we need to be careful. The solution would to _allow_ named expression and check if it's parenthesized or not.